src/util/suid.c

```diff
@@ -121,6 +113,13 @@ int singularity_suid_init(void) {
     }
 #endif /* SINGULARITY_SUID */
 
+    singularity_message(VERBOSE2, "Checking if singularity.conf allows us to run as suid\n");
+    if ( ( singularity_config_get_bool(ALLOW_SETUID) <= 0  ) || ( singularity_registry_get("NOSUID") != NULL ) ) {
+        envar_set("SINGULARITY_NOSUID", "1", 1);
+        singularity_registry_set("NOSUID", "1");
+        return(-1);
```

Comment:
Should we ABORT() if we can't initialize from this function properly?

Reply:
@gmkurtzer No because aborting there when 'allow setuid = no' will always fail, the return value is for condition https://github.com/cclerget/singularity/blob/670c701959286c77216d708576afd1b5da86d0b5/src/wrapper.c#L90 to run non setuid workflow

Reply:
@carlm1234 singularity_suid_init is called only in wrapper.c, the other files where this one appears are obsolete and not used anymore, no worries about that. To avoid confusion I will move this check into singularity_suid_allowed()